controllers/chat/initChatFromPost.js

```diff
@@ -42,35 +46,104 @@ const initChatFromPost = async (req, res) => {
+
+      await ChatAnonUserInfo.create({
+        channel_id: newChannel.id,
+        my_anon_user_id: userModel.user_id,
+        target_user_id: targetUserModel.user_id,
+        anon_user_info_color_code: postAnonUserInfo?.anon_user_info_color_code,
+        anon_user_info_color_name: postAnonUserInfo?.anon_user_info_color_name,
+        anon_user_info_emoji_code: postAnonUserInfo?.anon_user_info_emoji_code,
+        anon_user_info_emoji_name: postAnonUserInfo?.anon_user_info_emoji_name
+      });
+    }
+
+    if (userModel.is_anonymous) {
+      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_color_code = color.code;
+      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_color_name = color.color;
+      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_emoji_code = emoji.emoji;
+      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_emoji_name = emoji.name;
+
+      newStateMemberWithAnonInfo[userModel.user_id].user.name = 'Anonymous ' + emoji.name;
+      newStateMemberWithAnonInfo[userModel.user_id].user.username = 'Anonymous ' + emoji.name;
+
+      await ChatAnonUserInfo.create({
+        channel_id: newChannel.id,
+        my_anon_user_id: userModel.user_id,
+        target_user_id: targetUserModel.user_id,
+        anon_user_info_color_code: color.code,
+        anon_user_info_color_name: color.color,
+        anon_user_info_emoji_code: emoji.emoji,
+        anon_user_info_emoji_name: emoji.name
+      });
+    }
+    //end add anon info to channel
 
     try {
-      if (!channel?.data?.name) {
+      if (!newChannel?.data?.name) {
         let channelType = CHANNEL_TYPE.CHAT;
         if (userModel.is_anonymous || targetUserModel.is_anonymous) {
           channelType = CHANNEL_TYPE.ANONYMOUS;
         }
-
-        await channel.updatePartial({
+        await newChannel.updatePartial({
           set: {
             channel_type: channelType,
-            name: [userModel?.username, targetUserModel?.username].join(', ')
+            name: [userModel?.username, targetUserModel?.username].join(', '),
+            anon_user_info_color_code: color.code,
+            anon_user_info_color_name: color.color,
+            anon_user_info_emoji_code: emoji.emoji,
+            anon_user_info_emoji_name: emoji.name,
+            better_channel_member: newStateMemberWithAnonInfo
           }
         });
       }
     } catch (e) {
       console.log(e);
     }
 
+    const channelState = await newChannel.watch();
+
     const response = {
-      ...createdChannel,
-      better_channel_members_object: channel.state.members,
-      better_channel_members: Object.values(channel.state.members)
+      ...channelState,
+      better_channel_members: Object.values(channelState.channel.better_channel_member),
+      better_channel_members_objet: channelState.channel.better_channel_member,
+      messageHistories: []
     };
 
+    await newChannel.stopWatching();
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are a few issues in this hunk:

1. Error handling: There is no error handling for the `await newChannel.create();` operation on line 52. If an error occurs during channel creation, it will cause the application to crash. It's better to wrap this in a try-catch block.

2. Directly modifying user data: On lines 77-80 and 99-100, you're directly modifying the user's name and username. This could lead to inconsistencies if the user's actual name or username is needed elsewhere. Instead of directly modifying these properties, consider adding new properties to hold the anonymous name and username.

3. Logging errors: On line 133, you're logging the error to the console. While this is fine for development, in production, you should consider using a more robust logging solution that can provide more context about the error and support log levels.

Here are the suggested changes:

```diff
    const newChannel = client.channel('messaging', uuid(), {
       members
     });
- 52:     await newChannel.create();
+ 52:     try {
+ 53:       await newChannel.create();
+ 54:     } catch (error) {
+ 55:       // Log the error and return a response indicating the failure
+ 56:       console.error('Failed to create new channel:', error);
+ 57:       return res.status(500).json(new ErrorResponse('Failed to create new channel'));
+ 58:     }
    let newStateMemberWithAnonInfo = newChannel.state.members;

    //Add anon info to channel
    if (targetUserModel.is_anonymous) {
      const postAnonUserInfo = await PostAnonUserInfoFunction.checkSelfUsernameInPost(
        PostAnonUserInfo,
        User,
        {
          postId: postId,
          userId: targetUserModel.user_id
        },
        null,
        true
      );

      newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code =
        postAnonUserInfo?.anon_user_info_color_code;
      newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_name =
        postAnonUserInfo?.anon_user_info_color_name;
      newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_code =
        postAnonUserInfo?.anon_user_info_emoji_code;
      newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name =
        postAnonUserInfo?.anon_user_info_emoji_name;

- 83:       newStateMemberWithAnonInfo[targetUserModel.user_id].user.name =
- 84:         'Anonymous ' + postAnonUserInfo?.anon_user_info_emoji_name;
- 85:       newStateMemberWithAnonInfo[targetUserModel.user_id].user.username =
- 86:         'Anonymous ' + postAnonUserInfo?.anon_user_info_emoji_name;
+ 83:       newStateMemberWithAnonInfo[targetUserModel.user_id].user.anonymous_name =
+ 84:         'Anonymous ' + postAnonUserInfo?.anon_user_info_emoji_name;
+ 85:       newStateMemberWithAnonInfo[targetUserModel.user_id].user.anonymous_username =
+ 86:         'Anonymous ' + postAnonUserInfo?.anon_user_info_emoji_name;

      await ChatAnonUserInfo.create({
        channel_id: newChannel.id,
        my_anon_user_id: userModel.user_id,
        target_user_id: targetUserModel.user_id,
        anon_user_info_color_code: postAnonUserInfo?.anon_user_info_color_code,
        anon_user_info_color_name: postAnonUserInfo?.anon_user_info_color_name,
        anon_user_info_emoji_code: postAnonUserInfo?.anon_user_info_emoji_code,
        anon_user_info_emoji_name: postAnonUserInfo?.anon_user_info_emoji_name
      });
    }

    if (userModel.is_anonymous) {
      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_color_code = color.code;
      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_color_name = color.color;
      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_emoji_code = emoji.emoji;
      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_emoji_name = emoji.name;

- 105:       newStateMemberWithAnonInfo[userModel.user_id].user.name = 'Anonymous ' + emoji.name;
- 106:       newStateMemberWithAnonInfo[userModel.user_id].user.username = 'Anonymous ' + emoji.name;
+ 105:       newStateMemberWithAnonInfo[userModel.user_id].user.anonymous_name = 'Anonymous ' + emoji.name;
+ 106:       newStateMemberWithAnonInfo[userModel.user_id].user.anonymous_username = 'Anonymous ' + emoji.name;

      await ChatAnonUserInfo.create({
        channel_id: newChannel.id,
        my_anon_user_id: userModel.user_id,
        target_user_id: targetUserModel.user_id,
        anon_user_info_color_code: color.code,
        anon_user_info_color_name: color.color,
        anon_user_info_emoji_code: emoji.emoji,
        anon_user_info_emoji_name: emoji.name
      });
    }
    //end add anon info to channel
 
     try {
      if (!newChannel?.data?.name) {
         let channelType = CHANNEL_TYPE.CHAT;
         if (userModel.is_anonymous || targetUserModel.is_anonymous) {
           channelType = CHANNEL_TYPE.ANONYMOUS;
         }
        await newChannel.updatePartial({
           set: {
             channel_type: channelType,
            name: [userModel?.username, targetUserModel?.username].join(', '),
            anon_user_info_color_code: color.code,
            anon_user_info_color_name: color.color,
            anon_user_info_emoji_code: emoji.emoji,
            anon_user_info_emoji_name: emoji.name,
            better_channel_member: newStateMemberWithAnonInfo
           }
         });
       }
     } catch (e) {
- 139:        console.log(e);
+ 139:        console.error('Failed to update channel:', e);
     }
 
    const channelState = await newChannel.watch();

     const response = {
      ...channelState,
      better_channel_members: Object.values(channelState.channel.better_channel_member),
      better_channel_members_objet: channelState.channel.better_channel_member,
      messageHistories: []
     };
 
    await newChannel.stopWatching();

     await client.disconnectUser();
 
     return res.status(200).json(responseSuccess('sent', response));
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->